apollo/lib/websocket_manager.py

```diff
@@ -76,7 +76,10 @@ async def send_message_and_wait_for_response(
     async def close_and_remove_connection(self, websocket_id):
         websocket = self.connections.pop(websocket_id)
         try:
-            await websocket.send_json("Closing connection")
+            await websocket.send_json({
```

Comment:
There's no way to tell now if this went well before you close the connection in the next statement.

This class also currently is pretty specific to agent websocket connections but not explicitly so. This makes me question whether this class should be responsible for cleanup rather than just communication.